### DISCUSSION TITLE
Sniffing files in tests directories

### DISCUSSION THREAD

**Discussion by g-elwell** [bigbite/phpcs-config#166] [Ideas]
We're currently excluding files in `tests` directories from being sniffed. There's likely a good reason for this, and we may want to enforce a different ruleset for these files than within our application code.

However, without any sniffing whatsoever, it's easy to hit formatting issues which would otherwise be avoided.

Further to this there may be test-specific sniffs we could implement. I haven't looked into anything at this stage, but as there are conventions which need to be followed in relation to how test files and methods are named, PHPCS may be able to help us out here.

**Comment by jaymcp** [comment]
Thanks for opening this discussion.

There are some rules in our PHPCS config that would interfere with PHPUnit, so we exclude the directory to avoid developer frustration. However, I do understand that at least a subset of our rules would be useful to have apply to test cases. 

Perhaps, rather than excluding the tests directory completely, we could instead disable certain rules? 🤔

**Reply by g-elwell** [reply]
Do you recall which rules may have been interfering with PHPUnit previously?

I've just ran our ruleset against a repo with a large variety of PHPUnit integration and unit tests, and the only warnings I was seeing related to either file naming conventions or missing doc/function comments - neither of which impacted test behaviour once resolved.

**Reply by jaymcp** [reply]
That's great to hear!
I can't recall exactly which rules we have that caused issues, and it was a long time ago.
I'd be happy to remove the filter that prevents scanning test directories, but it would be a breaking change, so we'd have to major version bump.

**Reply by g-elwell** [reply]
I think it would be a worthwhile change, especially considering you can exclude directories at the project level, but cannot revert the exclusion applied within this config
